rfd/0008-application-access.md

```diff
@@ -0,0 +1,109 @@
+
+We have provided an example within Teleport on how to [validate the JWT token](https://github.com/gravitational/teleport/blob/a7fca6bf2bc15fc487620b410151c2108ac1d6cc/lib/jwt/jwt.go#L192-L225) written in Go.
+
+#### Claims
+
+The JWT embeds within it claims about the identity of the subject and issuer of the token.
+
+The following public claims are included:
+
+* `aud`: Audience of JWT. This is the URI of the proxied application to which the request is being forwarded.
+* `exp`: Expiration time of the JWT. This value is always in sync with the expiration of the TLS certificate.
+* `iss`: Issuer of the JWT. This value is the name of the Teleport cluster issuing the token.
+* `nbf`: Not before time of the JWT. This is the time at which the JWT becomes valid.
+* `sub`: Subject of the JWT. This is the Teleport identity of the user to whom the JWT was issued.
+
+The following private claims are included.
+
+* `username`: Similar to sub. This is the Teleport identity of the user to whom the JWT was issued.
+* `roles`: List of Teleport roles assigned to the user.
+
+#### Rotation
+
+The JWT signing keys are rotated along with User and Host CAs when using the `tctl auth rotate [...]` command. If you specifically only want to rotate your JWT signer, use the `--type=jwt` flag.
+
+#### Example
+
+The following header will be sent to an internal application:
+
+```
+teleport-jwt-assertion: eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsiaHR0cDovL2xvY2FsaG9zdDo4MDgwL2FwcCJdLCJleHAiOjE2MDM1Mjk5NzEsImlzcyI6ImV4YW1wbGUuY29tIiwibmJmIjoxNjAzNDg2Nzg2LCJyb2xlcyI6WyJhZG1pbiJdLCJzdWIiOiJyam9uZXMiLCJ1c2VybmFtZSI6InJqb25lcyJ9.SnyYMyjjcxEUsPnf-WxWy33yVWsHR3hQPCml-fizX1HJY7jojkKroPbrXBCO-WEJ8RqCzv0j6u1pz_PllNPhhPrCE8Q32WAB2OaazVM2FHxsiEVyMInCUVEAsrieYo0BQidXTj85yGgEPV45VdbnqWdJSzVr1UmUF6kDdMwhS3Zyr-SRAZw9ix_jBK6nxDmlD0TgJh9eAvRhbjvxU12I6A4VqZVPTrefoWsdZTHrYvg2oqztHtNSycqsbqfIBnNmg__opWKgouW_t-Xv58aA8scW_5DavVitPhQBbsPH0QRKfu-xMNDtmfa6eBAKe7E9uO2uDcmDA26dHKIA2n90Gw
+```
+
+Decoding to the below JSON.
+
+```
+{
+  "aud": [
+    "http://localhost:8080/app"
+  ],
+  "exp": 1603530049,
+  "iss": "example.com",
+  "nbf": 1603486842,
+  "roles": [
+    "admin"
+  ],
+  "sub": "foo",
+  "username": "foo"
+}
+```
+
+### Logout
+
+Each application the user launches maintains its own session. Sessions automatically TTL out after the time specified on the role and certificate.
+
+To explicitly logout a session, an authenticated session can issue a `GET /teleport-logout` or a `DELETE /teleport-logout` request.
+
+Internal applications and implementers are encouraged to support `DELETE /teleport-logout` in the form of a logout button within the internal application.
+
+The `GET /teleport-logout` endpoint is for internal applications that can not be modified.
```

Comment:
So this we be a practical example to logout?
 
Vist `https://[teleport-application-access-public-addr]/teleport-logout`

Reply:
Yes.